test/xbgpu/test_engine.py

```diff
@@ -1289,42 +1370,57 @@ def heap_factory(batch_index: int, present: np.ndarray) -> list[spead2.send.Heap
             assert stream_diff.diff("output_x_clipped_visibilities_total") == n_vis
             assert xbengine.sensors[f"{corrprod_output.name}.xeng-clip-cnt"].value == n_vis
 
-    def _patch_get_in_item(
+    def _patch_method(
         self,
         monkeypatch: pytest.MonkeyPatch,
+        class_or_instance: object,
+        method_name: str,
         count: int,
         client: aiokatcp.Client,
         requests: Iterable[Iterable],
     ) -> list[int]:
-        """Patch :meth:`~.BPipeline._get_in_item` to make requests partway through the stream.
+        """Patch `method_name` of class `class_type` to make requests during operation.
```

Comment:
Lingering reference to `class_type`.

`class_or_instance` is a bit of a mouthful. I'd just call it `obj` (since *any* object is a class or instance), and `obj` is the [parameter name](https://docs.pytest.org/en/stable/how-to/monkeypatch.html#how-to-monkeypatch-mock-modules-and-environments) used by monkeypatch.

It is probably worth noting in the body of the docstring that it can be used to patch both classes and instances though.

Reply:
Ah, fair enough. I had a peek at the type hint monkeypatch uses. Didn't go deep enough for the parameter name.